src/IB/IIMethod.cpp

```diff
@@ -232,6 +232,126 @@ IIMethod::getFEDataManager(const unsigned int part) const
     TBOX_ASSERT(part < d_num_parts);
     return d_fe_data_managers[part];
 } // getFEDataManager
+NumericVector<double>* 
+IIMethod::getMeshCoordinatesNumeric(bool isCurrentConfiguration,std::string time,unsigned int part){
+
+
+    if(isCurrentConfiguration){
+
+        NumericVector<double>* X_ghost_vec = d_X_IB_ghost_vecs[part];
+        NumericVector<double>* X_vec = nullptr;
+
+        if(time == "current"){
+            X_vec = d_X_current_vecs[part];
+            copy_and_synch(*X_vec, *X_ghost_vec); 
+
+        }
+        else if(time == "half"){
+            X_vec = d_X_half_vecs[part];
+            copy_and_synch(*X_vec, *X_ghost_vec); 
+        }
+        else if(time == "new"){
+            X_vec = d_X_new_vecs[part];
+            copy_and_synch(*X_vec, *X_ghost_vec); 
+        }
+        else if(time == "ib_ghost"){
+            //already assigned X_ghost_veec
+        }
+        else{
+            TBOX_ERROR("time must be set to current, half, new, or ib_ghost. case sensitive.");
+        }
+        return X_ghost_vec;
+
+    }
+    else{ //reference configuration
+
+        EquationSystems* equation_systems = d_fe_data_managers[part]->getEquationSystems();
+        const MeshBase& mesh = equation_systems->get_mesh();
+        const unsigned int dim = mesh.mesh_dimension();
+        std::unique_ptr<QBase> qrule;
+        System& X_system = equation_systems->get_system(COORDS_SYSTEM_NAME);
+        const DofMap& X_dof_map = X_system.get_dof_map();
+        FEDataManager::SystemDofMapCache& X_dof_map_cache =
+            *d_fe_data_managers[part]->getDofMapCache(COORDS_SYSTEM_NAME);
+        FEType X_fe_type = X_dof_map.variable_type(0);
+
+        //NumericVector<double>* X_ghost_vec = d_X_IB_ghost_vecs[part];
+        //X_ghost_vec->close();
+        //PetscVector<double>* X_petsc_vec = static_cast<PetscVector<double>*>(X_ghost_vec);  
+        //NumericVector<double>* X0_vec;
+        //std::unique_ptr<NumericVector<double> > X0_vec = X_petsc_vec->clone();
+        //copy_and_synch(X_system.get_vector("INITIAL_COORDINATES"), *X0_vec);
```

Comment:
Please remove commented-out code for PRs that are intended to be merged with `master`.